jax/_src/numpy/lax_numpy.py

```diff
@@ -1733,6 +1733,7 @@ def bincount(x, weights=None, minlength=0, *, length=None):
     raise ValueError("shape of weights must match shape of x.")
   return zeros(length, _dtype(weights)).at[clip(x, 0)].add(weights)
 
+broadcast_shapes = _wraps(getattr(np, "broadcast_shapes", None))(lax.broadcast_shapes)
```

Comment:
I think we'll need a bit more of a wrapper than this to match numpy's behavior: `lax.broadcast_shapes` does not accept integers, while numpy's does:
```python
In [1]: import numpy as np                                                                                                                              

In [2]: np.broadcast_shapes(3, (2, 1))                                                                                                                  
Out[2]: (2, 3)

In [3]: from jax import lax                                                                                                                             

In [4]: lax.broadcast_shapes(3, (2, 1))                                                                                                                 
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-4-0981db21d119> in <module>
----> 1 lax.broadcast_shapes(3, (2, 1))

~/github/google/jax/jax/_src/util.py in wrapper(*args, **kwargs)
    184         return f(*args, **kwargs)
    185       else:
--> 186         return cached(config._trace_context(), *args, **kwargs)
    187 
    188     wrapper.cache_clear = cached.cache_clear

~/github/google/jax/jax/_src/util.py in cached(_, *args, **kwargs)
    177     @functools.lru_cache(max_size)
    178     def cached(_, *args, **kwargs):
--> 179       return f(*args, **kwargs)
    180 
    181     @functools.wraps(f)

~/github/google/jax/jax/_src/lax/lax.py in broadcast_shapes(*shapes)
     86   if len(shapes) == 1:
     87     return shapes[0]
---> 88   ndim = _max(len(shape) for shape in shapes)
     89   shapes = [(1,) * (ndim - len(shape)) + shape for shape in shapes]
     90   result_shape = _try_broadcast_shapes(shapes)

~/github/google/jax/jax/_src/lax/lax.py in <genexpr>(.0)
     86   if len(shapes) == 1:
     87     return shapes[0]
---> 88   ndim = _max(len(shape) for shape in shapes)
     89   shapes = [(1,) * (ndim - len(shape)) + shape for shape in shapes]
     90   result_shape = _try_broadcast_shapes(shapes)

TypeError: object of type 'int' has no len()
```

Reply:
Good catch! I forgot about that.
I updated the PR and adapted `lax.broadcast_shapes` to handle integers, lists and numpy arrays as shapes. Let me know if you prefer to move this handling to the numpy function instead.